[R-package] Fix Rmarkdown compilation

This might due to recent updates in interface, there are a few errors in the rmarkdown compilation here https://xgboost.readthedocs.org/en/latest/R-package/index.html

@pommedeterresautee can you look into it and update these. See how to compile the rmarkdown in https://xgboost.readthedocs.org/en/latest/dev-guide/contribute.html#r-package

@tqchen I have some difficulties to compile the project on my Windows 8 machine.

```
==> devtools::document(roclets=c('rd', 'collate', 'namespace'))

Updating xgboost documentation
Loading xgboost
Re-compiling xgboost
"C:/PROGRA~1/R/R-32~1.3/bin/i386/R" --no-site-file --no-environ --no-save  \
  --no-restore CMD INSTALL  \
  "C:\Users\mbenesty\Documents\GitHub\xgboost\R-package"  \
  --library="C:\Users\mbenesty\AppData\Local\Temp\Rtmp6LJMlg\devtools_install_19bc14b63144"  \
  --no-R --no-data --no-help --no-demo --no-inst --no-docs --no-exec  \
  --no-multiarch --no-test-load --preclean 

* installing *source* package 'xgboost' ...
** libs
cp -r ../../src .
cp -r ../../rabit .
cp -r ../../dmlc-core .
cp -r ../../include .
cp -r ../../amalgamation .
g++ -m32 -std=c++0x -I"C:/PROGRA~1/R/R-32~1.3/include" -DNDEBUG -I.//include -I.//dmlc-core/include -I.//rabit/include -I./ -DXGBOOST_STRICT_R_MODE=1 -DDMLC_LOG_BEFORE_THROW=0 -DDMLC_ENABLE_STD_THREAD=0 -DDMLC_DISABLE_STDIN=1 -DDMLC_LOG_CUSTOMIZE=1 -DXGBOOST_CUSTOMIZE_LOGGER=1 -DRABIT_CUSTOMIZE_MSG_ -DRABIT_STRICT_CXX98_    -I"d:/RCompile/r-compiling/local/local323/include"  -fopenmp -pthread   -O2 -Wall  -mtune=core2 -c xgboost_R.cc -o xgboost_R.o
xgboost_R.cc:2:26: fatal error: dmlc/logging.h: No such file or directory
compilation terminated.
make: *** [xgboost_R.o] Error 1
Warning: l'ex�cution de la commande 'make -f "Makevars.win" -f "C:/PROGRA~1/R/R-32~1.3/etc/i386/Makeconf" -f "C:/PROGRA~1/R/R-32~1.3/share/make/winshlib.mk" CXX='$(CXX1X) $(CXX1XSTD)' CXXFLAGS='$(CXX1XFLAGS)' CXXPICFLAGS='$(CXX1XPICFLAGS)' SHLIB_LDFLAGS='$(SHLIB_CXX1XLDFLAGS)' SHLIB_LD='$(SHLIB_CXX1XLD)' SHLIB="xgboost.dll" ' renvoie un statut 2
ERROR: compilation failed for package 'xgboost'
* removing 'C:/Users/mbenesty/AppData/Local/Temp/Rtmp6LJMlg/devtools_install_19bc14b63144/xgboost'
Erreur : Command failed (1)
Ex�cution arr�t�e

Exited with status 1.
```

I will try this sunday on my Linux machine.

Kind regards,
Michaël

I encountered the same error: 
```
* installing *source* package ‘xgboost’ ...
** libs
clang++ -std=c++11 -I/usr/local/Cellar/r/3.2.1_1/R.framework/Resources/include -DNDEBUG -I../..//include -I../..//dmlc-core/include -I../..//rabit/include -I../../ -DXGBOOST_STRICT_R_MODE=1 -DDMLC_LOG_BEFORE_THROW=0 -DDMLC_ENABLE_STD_THREAD=1 -DDMLC_DISABLE_STDIN=1 -DDMLC_LOG_CUSTOMIZE=1 -DXGBOOST_CUSTOMIZE_LOGGER=1 -DRABIT_CUSTOMIZE_MSG_ -DRABIT_STRICT_CXX98_ -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include     -fPIC  -g -O2 -c xgboost_R.cc -o xgboost_R.o
xgboost_R.cc:2:10: fatal error: 'dmlc/logging.h' file not found
#include <dmlc/logging.h>
         ^
1 error generated.
make: *** [xgboost_R.o] Error 1
ERROR: compilation failed for package ‘xgboost’
* removing ‘/usr/local/Cellar/r/3.2.1_1/R.framework/Versions/3.2/Resources/library/xgboost’
* restoring previous ‘/usr/local/Cellar/r/3.2.1_1/R.framework/Versions/3.2/Resources/library/xgboost’
Error: Command failed (1)
```
@tqchen Last time I was executing the same `devtools::install_git('git://github.com/dmlc/xgboost',subdir='R-package')` but it's actually installing the version on master so the installation actually didn't succeed on mac. Maybe there's something I missed.

I am having the same compilation error, @tqchen , I tried your solution without succes.

```R
devtools::install_git('git://github.com/dmlc/xgboost',subdir='R-package', args='--recursive')
Downloading git repo git://github.com/dmlc/xgboost
Installing xgboost
'/usr/lib64/R/bin/R' --no-site-file --no-environ --no-save --no-restore CMD INSTALL '/tmp/Rtmp1jStfU/file3e382ec8d331/R-package'  \
  --library='/home/sam/R/x86_64-pc-linux-gnu-library/3.2' --install-tests 

* installing *source* package ‘xgboost’ ...
** libs
g++ -std=c++11 -I/usr/include/R/ -DNDEBUG -I../..//include -I../..//dmlc-core/include -I../..//rabit/include -I../../ -DXGBOOST_STRICT_R_MODE=1 -DDMLC_LOG_BEFORE_THROW=0 -DDMLC_ENABLE_STD_THREAD=1 -DDMLC_DISABLE_STDIN=1 -DDMLC_LOG_CUSTOMIZE=1 -DXGBOOST_CUSTOMIZE_LOGGER=1 -DRABIT_CUSTOMIZE_MSG_ -DRABIT_STRICT_CXX98_ -D_FORTIFY_SOURCE=2   -fopenmp  -fpic  -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -c xgboost_R.cc -o xgboost_R.o
xgboost_R.cc:2:26: fatal error: dmlc/logging.h: No such file or directory
compilation terminated.
/usr/lib64/R/etc/Makeconf:141: recipe for target 'xgboost_R.o' failed
make: *** [xgboost_R.o] Error 1
ERROR: compilation failed for package ‘xgboost’
* removing ‘/home/sam/R/x86_64-pc-linux-gnu-library/3.2/xgboost’
* restoring previous ‘/home/sam/R/x86_64-pc-linux-gnu-library/3.2/xgboost’
Error: Command failed (1)
In addition: Warning message:
`args` is deprecated 
````

Maybe the recursive argument isn't passed to git?

--- a/R-package/vignettes/xgboost.Rnw
+++ b/R-package/vignettes/xgboost.Rnw
@@ -21,7 +21,7 @@ if (require('knitr')) opts_chunk$set(fig.width = 5, fig.height = 5, fig.align = 
 
 %
 <<prelim,echo=FALSE>>=
-xgboost.version = '0.3-0'
+xgboost.version = '0.4-2'
 @
 %
 

--- a/R-package/vignettes/xgboostPresentation.Rmd
+++ b/R-package/vignettes/xgboostPresentation.Rmd
@@ -49,18 +49,24 @@ It has several features:
 ### Github version
 
 
-For up-to-date version (highly recommended), install from *Github*:
+For weekly updated version (highly recommended), install from *Github*:
 
 ```{r installGithub, eval=FALSE}
-devtools::install_git('git://github.com/dmlc/xgboost', subdir='R-package')
+install.packages("drat", repos="https://cran.rstudio.com")
+drat:::addRepo("dmlc")
+install.packages("xgboost", repos="http://dmlc.ml/drat/", type = "source")
 ```
 
-> *Windows* user will need to install [RTools](http://cran.r-project.org/bin/windows/Rtools/) first.
+> *Windows* user will need to install [Rtools](http://cran.r-project.org/bin/windows/Rtools/) first.
 
-Cran version
-------------
+### CRAN version
 
-As of 2015-03-13, ‘xgboost’ was removed from the CRAN repository.
+
+The version 0.4-2 is on CRAN, and you can install it by:
+
+```{r, eval=FALSE}
+install.packages("xgboost")
+```
 
 Formerly available versions can be obtained from the CRAN [archive](http://cran.r-project.org/src/contrib/Archive/xgboost)
 

--- a/doc/R-package/Makefile
+++ b/doc/R-package/Makefile
@@ -8,6 +8,7 @@ xgboostPresentation.md: $(PKGROOT)/vignettes/xgboostPresentation.Rmd
 # General Rules for build rmarkdowns, need knitr
 %.md:
 	Rscript -e \
+	"require(methods);"\
 	"require(knitr);"\
 	"knitr::opts_knit\$$set(root.dir=\".\");"\
 	"knitr::opts_chunk\$$set(fig.path=\"../web-data/xgboost/knitr/$(basename $@)-\");"\

--- a/doc/R-package/xgboostPresentation.md
+++ b/doc/R-package/xgboostPresentation.md
@@ -49,19 +49,26 @@ It has several features:
 ### Github version
 
 
-For up-to-date version (highly recommended), install from *Github*:
+For weekly updated version (highly recommended), install from *Github*:
 
 
 ```r
-devtools::install_git('git://github.com/dmlc/xgboost', subdir='R-package')
+install.packages("drat", repos="https://cran.rstudio.com")
+drat:::addRepo("dmlc")
+install.packages("xgboost", repos="http://dmlc.ml/drat/", type = "source")
 ```
 
-> *Windows* user will need to install [RTools](http://cran.r-project.org/bin/windows/Rtools/) first.
+> *Windows* user will need to install [Rtools](http://cran.r-project.org/bin/windows/Rtools/) first.
 
-Cran version
-------------
+### CRAN version
 
-As of 2015-03-13, ‘xgboost’ was removed from the CRAN repository.
+
+The version 0.4-2 is on CRAN, and you can install it by:
+
+
+```r
+install.packages("xgboost")
+```
 
 Formerly available versions can be obtained from the CRAN [archive](http://cran.r-project.org/src/contrib/Archive/xgboost)
 
@@ -115,11 +122,16 @@ str(train)
 
 ```
 ## List of 2
-##  $ data :
-```
-
-```
-## Error in str.default(obj, ...): could not find function "is"
+##  $ data :Formal class 'dgCMatrix' [package "Matrix"] with 6 slots
+##   .. ..@ i       : int [1:143286] 2 6 8 11 18 20 21 24 28 32 ...
+##   .. ..@ p       : int [1:127] 0 369 372 3306 5845 6489 6513 8380 8384 10991 ...
+##   .. ..@ Dim     : int [1:2] 6513 126
+##   .. ..@ Dimnames:List of 2
+##   .. .. ..$ : NULL
+##   .. .. ..$ : chr [1:126] "cap-shape=bell" "cap-shape=conical" "cap-shape=convex" "cap-shape=flat" ...
+##   .. ..@ x       : num [1:143286] 1 1 1 1 1 1 1 1 1 1 ...
+##   .. ..@ factors : list()
+##  $ label: num [1:6513] 1 0 0 1 0 0 0 1 0 0 ...
 ```
 
 `label` is the outcome of our dataset meaning it is the binary *classification* we will try to predict.
@@ -206,7 +218,8 @@ bstDense <- xgboost(data = as.matrix(train$data), label = train$label, max.depth
 ```
 
 ```
-## Error in as.vector(data): no method for coercing this S4 class to a vector
+## [0]	train-error:0.046522
+## [1]	train-error:0.022263
 ```
 
 ##### xgb.DMatrix
@@ -254,9 +267,9 @@ bst <- xgboost(data = dtrain, max.depth = 2, eta = 1, nthread = 2, nround = 2, o
 ```
 
 ```
-## [11:43:20] ../..//amalgamation/../src/tree/updater_prune.cc:74: tree pruning end, 1 roots, 6 extra nodes, 0 pruned nodes, max_depth=2
+## [11:41:01] amalgamation/../src/tree/updater_prune.cc:74: tree pruning end, 1 roots, 6 extra nodes, 0 pruned nodes, max_depth=2
 ## [0]	train-error:0.046522
-## [11:43:20] ../..//amalgamation/../src/tree/updater_prune.cc:74: tree pruning end, 1 roots, 4 extra nodes, 0 pruned nodes, max_depth=2
+## [11:41:01] amalgamation/../src/tree/updater_prune.cc:74: tree pruning end, 1 roots, 4 extra nodes, 0 pruned nodes, max_depth=2
 ## [1]	train-error:0.022263
 ```
 
@@ -412,8 +425,8 @@ bst <- xgb.train(data=dtrain, booster = "gblinear", max.depth=2, nthread = 2, nr
 ```
 
 ```
-## [0]	train-error:0.019499	train-logloss:0.176561	test-error:0.018001	test-logloss:0.173835
-## [1]	train-error:0.004760	train-logloss:0.068214	test-error:0.003104	test-logloss:0.065493
+## [0]	train-error:0.024720	train-logloss:0.184616	test-error:0.022967	test-logloss:0.184234
+## [1]	train-error:0.004146	train-logloss:0.069885	test-error:0.003724	test-logloss:0.068081
 ```
 
 In this specific case, *linear boosting* gets sligtly better performance metrics than decision trees based algorithm.
@@ -442,7 +455,7 @@ dtrain2 <- xgb.DMatrix("dtrain.buffer")
 ```
 
 ```
-## [11:43:20] 6513x126 matrix with 143286 entries loaded from dtrain.buffer
+## [11:41:01] 6513x126 matrix with 143286 entries loaded from dtrain.buffer
 ```
 
 ```r

--- a/doc/build.md
+++ b/doc/build.md
@@ -143,10 +143,12 @@ There are several ways to install the package:
 
 ## R Package Installation
 
-You can install R package using devtools
+You can install R package from cran just like other packages, or you can install from our weekly updated drat repo:
 
 ```r
-devtools::install_git('git://github.com/dmlc/xgboost',subdir='R-package')
+install.packages("drat", repos="https://cran.rstudio.com")
+drat:::addRepo("dmlc")
+install.packages("xgboost", repos="http://dmlc.ml/drat/", type = "source")
 
 ```
 
@@ -160,13 +162,9 @@ First follow [Building on OSX](#building-on-osx) to get the OpenMP enabled compi
   Then inside R, run
 
   ```R
-  install.packages('xgboost/R-package/', repos=NULL, type='source')
-  ```
-
-  Or
-
-  ```R
-  devtools::install_local('xgboost/', subdir = 'R-package') # you may use devtools
+  install.packages("drat", repos="https://cran.rstudio.com")
+  drat:::addRepo("dmlc")
+  install.packages("xgboost", repos="http://dmlc.ml/drat/", type = "source")
   ```
 
 ## Trouble Shooting
@@ -198,4 +196,3 @@ First follow [Building on OSX](#building-on-osx) to get the OpenMP enabled compi
     ```bash
     git clone https://github.com/dmlc/xgboost --recursive
     ```
-    

